embedded-graphics/src/lib.rs

```diff
@@ -232,18 +237,130 @@ use crate::pixelcolor::PixelColor;
+///
+/// In addition to defining `draw_pixel`, an implementation of `DrawTarget` can also provide
+/// alternative implementations of drawing methods for graphical primitives. Here is an example of
+/// how a display with accelerated methods can implement `DrawTarget`:
+///
+/// ```rust
+/// # use embedded_graphics::prelude::*;
+/// # use embedded_graphics::DrawTarget;
+/// # use embedded_graphics::egrectangle;
+/// # use embedded_graphics::primitives::rectangle::Rectangle;
+/// # use embedded_graphics::pixelcolor::{Gray8, GrayColor};
+/// # use embedded_graphics::drawable::Pixel;
+/// #
+/// # struct SPI1;
+/// #
+/// # impl SPI1 {
+/// #     pub fn send_bytes(&self, buf: &[u8]) -> Result<(), ()> {
+/// #         Ok(())
+/// #     }
+/// # }
+/// #
+/// /// A fake display 64px x 64px where each pixel is stored as a single `u8`
+/// struct FastExampleDisplay {
+///     framebuffer: [u8; 64 * 64],
+///     iface: SPI1,
+/// }
+///
+/// impl FastExampleDisplay {
+///     /// Send buffer to the display
+///     pub fn flush(&self) -> Result<(), ()> {
+///         self.iface.send_bytes(&self.framebuffer)
+///     }
+///
+///     /// A HW-accelerated method for drawing rectangles
+///     pub fn fast_rectangle(&self, rect: &Rectangle<Gray8>) {
+///         // Does some speedy drawing
+///     }
+/// }
+///
+/// impl DrawTarget<Gray8> for FastExampleDisplay {
+///     /// Draw a `pixel` that has a colour defined as `Gray8`
+///     fn draw_pixel(&mut self, pixel: Pixel<Gray8>) {
+///         let Pixel(coord, color) = pixel;
+///         // Place an (x, y) pixel at the right index in the framebuffer
+///         let index = coord[0] + (coord[1] * 64);
+///         self.framebuffer[index as usize] = color.luma();
+///     }
+///
+///     /// Use the accelerated method when drawing rectangles
+///     fn draw_rectangle(&mut self, item: &Rectangle<Gray8>) {
+///         self.fast_rectangle(item);
+///     }
+/// }
+///
+/// fn main() {
+///     let mut display = FastExampleDisplay {
+///         framebuffer: [0; 4096],
+///         iface: SPI1
+///     };
+///
+///     // Draw a rectangle from (10, 20) to (30, 40) with a white stroke
+///     let mut rect = egrectangle!((10, 20), (30, 40), stroke_color = Some(Gray8::WHITE));
+///     rect.draw(&mut display); // Uses the accelerated draw_rectangle function
+///
+///     // Update the display
+///     display.flush().expect("Failed to send data to display");
+/// }
+///
+/// ```
+pub trait DrawTarget<C>
 where
     C: PixelColor,
 {
+    /// Draw a pixel on the display
+    fn draw_pixel(&mut self, item: drawable::Pixel<C>);
+
     /// Draw an object from an iterator over its pixels
-    fn draw<T>(&mut self, item: T)
+    fn draw_iter<T>(&mut self, item: T)
     where
-        T: IntoIterator<Item = drawable::Pixel<C>>;
+        T: IntoIterator<Item = drawable::Pixel<C>>,
+    {
+        for pixel in item {
+            self.draw_pixel(pixel);
+        }
+    }
+
+    /// Draw a line primitive.
```

Comment:
Nit

```suggestion
    /// Draw a line primitive
```

@rfuest likes full stops but the other doc comments on this trait don't have them so let's keep this consistent

Reply:
@rfuest likes to follow: https://doc.rust-lang.org/1.0.0/style/style/comments.html#sentence-structure

Reply:
Ah, makes sense! We should do a docs pass before 0.6.0 to make everything consistent

Reply:
#189 :)